### PR TITLE
Support chocolatey 0.9.9

### PR DESCRIPTION
**Pull request by rusitschka**
These changes make the cookbook work with chocolatey 0.9.9 (tested up to version 0.9.9.2). Maybe it still works with versions < 0.9.9 (not tested).

### CODE REVIEW

File: providers/default.rb [hunk 1]
```diff
@@ -107,25 +107,29 @@ def upgradeable?(name)
   end
 
   Chef::Log.debug("Checking to see if this chocolatey package is installed/upgradable: '#{name}'")
-  cmd = Mixlib::ShellOut.new("#{::ChocolateyHelpers.chocolatey_executable} version #{name} #{cmd_args}")
+  cmd = Mixlib::ShellOut.new("#{::ChocolateyHelpers.chocolatey_executable} upgrade -r --noop #{cmd_args} #{name}")
   cmd.run_command
-  !cmd.stdout.include?('Latest version installed')
+  result = cmd.stdout.chomp
+  return false if result.empty? # for chocolatey 0.9.9 + 0.9.9.1
```
Comment (ferventcoder) [inline providers/default.rb]: fixed for 0.9.9.2 - you saw that though... apparently. :)

Comment (rusitschka) [inline providers/default.rb]: Do you have any estimate date for a merge? The cookbook runs fine here with approximately 40 machines.

Comment (aaronsky) [inline providers/default.rb]: Can we please get this merged? This is a blocker on my workflow and I'd rather use the master branch if at all possible

Comment (ferventcoder) [inline providers/default.rb]: @guilhem @mwrock ^^ thoughts and feedback here? I'd press go, but I'd like you to have a run through to verify the changes.

Comment (guilhem) [inline providers/default.rb]: I'm quite ok with this PR.
But I would like to remove specific task for 0.9.9.(0|1) as this is "deprecated".
:)

And to be sure. This PR breaks compatibility with previous chocolatey versions?

Comment (ferventcoder) [inline providers/default.rb]: Yes. I went about it a bit differently with the Puppet provider where it would work with both.

Comment (guilhem) [inline providers/default.rb]: I don't have any big problem about being compatible only with a newer version of chocolatey.

Any user who want the "old" behaviour can pin his cookbook (And he really should use pessimistic versionning!)

We only have to push a new cookbook with minor updated

Comment (ferventcoder) [inline providers/default.rb]: Cool - then you will take this forward then? :+1:

Comment (rusitschka) [inline providers/default.rb]: We can also tag the current master branch - or branch it before the merge - to support the pre 0.9.9 versions. I think supporting both versions just results in more code - which decreases maintainability.

**Comment by swathe** [comment]
Any update on a merge for this?

**Comment by swathe** [comment]
I implemented the changes in this pull request in my local environment and everything is working perfectly.

**Comment by stevenelder** [comment]
Any blockers to merging this pull request?

**Comment by guilhem** [comment]
squashing + removing custom rules for 0.9.9 and 0.9.9.1 :)

**Comment by akaias** [comment]
Is there any ETA of this in the repository ?

**Comment by Scythril** [comment]
While this is being worked on, is there a way to manually install 0.9.8? I just spun up a new server which installed 0.9.9, and the Chef run is broken.

**Comment by ferventcoder** [comment]
https://chocolatey.org/install-lastposhclient.ps1

**Comment by cmluciano** [comment]
This looks good to me also. I just tested and tried to install the python2 package afterwards with success. At the moment, we are just using execute to install packages as a work-around.

**Comment by guilhem** [comment]
fixed by 8ea307f01f5d0b30cee9df45fbc9cc7e5a351b03

**Pull request closed without merging**